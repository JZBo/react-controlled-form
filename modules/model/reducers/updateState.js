/* @flow */
import type { Action } from '../../../types/Action'

export default function updateField(
  state: Object,
  { payload: { formId, newState } }: Action
): Object {
  return {
    ...state,
    [formId]: {
      ...state[formId],
      state: {
        ...state[formId].state,
        ...newState,
      },
    },
  }
}
