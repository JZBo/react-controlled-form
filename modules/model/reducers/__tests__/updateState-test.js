import updateState from '../updateState'

const payload = p => ({ payload: p })
const initialFormState = {
  data: {},
  state: { is_checked: false, another_valid: true },
}

describe('Updating form state', () => {
  it('should merge the state', () => {
    expect(
      updateState(
        { foo: initialFormState },
        payload({
          formId: 'foo',
          newState: { is_checked: true, something_valid: false },
        })
      )
    ).toMatchSnapshot()
  })
})
