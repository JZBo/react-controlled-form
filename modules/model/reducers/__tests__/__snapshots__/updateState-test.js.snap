// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Updating form state should merge the state 1`] = `
Object {
  "foo": Object {
    "data": Object {},
    "state": Object {
      "another_valid": true,
      "is_checked": true,
      "something_valid": false,
    },
  },
}
`;
